src/app/src/pages/Profile.tsx

```diff
@@ -152,6 +160,35 @@ export default function Profile({ currentUser }: Props): JSX.Element {
     fetchData();
   }, [author?.id]);
 
+  // Get Current User's followers and update isFollwing is the author is being followed by the current user
+  const [currentUserFollowers, setCurrentUserFollowers] = useState<Author[]>(
+    []
+  );
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const res = api.authors.withId("" + currentUser?.id).followings.list();
+        res.then((followers) => {
+          setCurrentUserFollowers(followers);
+        });
+        let isFollowing = false;
+        if (currentUserFollowers) {
+          for (let i = 0; i < currentUserFollowers.length; i++) {
+            if (currentUserFollowers[i].id === author?.id) {
+              isFollowing = true;
+              break;
+            }
+          }
+
+          setFollowing(isFollowing);
+        }
+      } catch (err) {
+        console.log(err);
+      }
+    }
+    fetchData();
+  }, [author?.id, currentUser?.id, currentUserFollowers]);
```

Comment:
Having currentUserFollowers in here causes infinite api calls (bad).
With this setup, I think the only option is to remove the dependency array completely.
This will result in the following list not updating immediately however.

If you look above and how profilePosts handles this issue (using a bool in lieu of the array) and find a way to move the initial button set up (the for loop you had) outside of the useEffect then it may work.

Reply:
using API call to determine if the current user is following so that dependency is removed

Reply:
AH yes sorry had this open for a while so didn't see kyles comments!